### PR TITLE
Fix: FWHGCRecHitProxyBuilder

### PR DESCRIPTION
**Pull request by Mourtz**
HGCRecHitProxy now renders the rechits with the correct geometry.
Also added Layer, Heatmap and Endcap selector.

@rovere

### CODE REVIEW

File: Fireworks/Calo/plugins/FWHGCRecHitProxyBuilder.cc [hunk 1]
```diff
@@ -1,60 +1,153 @@
...
 
    REGISTER_PROXYBUILDER_METHODS();
-
 private:
    FWHGCRecHitProxyBuilder( const FWHGCRecHitProxyBuilder& ) = delete;
    const FWHGCRecHitProxyBuilder& operator=( const FWHGCRecHitProxyBuilder& ) = delete;
-};
 
-REGISTER_FWPROXYBUILDER( FWHGCRecHitProxyBuilder, HGCRecHitCollection, "HGC RecHit", FWViewType::kISpyBit );
+   static const uint8_t gradient_steps = 9;
```
Comment (Dr15Jones) [inline Fireworks/Calo/plugins/FWHGCRecHitProxyBuilder.cc]: Best to be `static constexpr`

File: Fireworks/Calo/plugins/FWHGCRecHitProxyBuilder.cc [hunk 2]
```diff
@@ -1,60 +1,153 @@
...
+   const float gradient[3][gradient_steps] = {
```
Comment (Dr15Jones) [inline Fireworks/Calo/plugins/FWHGCRecHitProxyBuilder.cc]: Please make `static constexpr float`

**Review by Dr15Jones — CHANGES_REQUESTED** [review]

**Comment by fabiocos** [comment]
please test

**Comment by alja** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**